pkg/apis/pipeline/v1beta1/taskrun_types.go

```diff
@@ -102,6 +98,43 @@ type TaskRunStatus struct {
 	TaskRunStatusFields `json:",inline"`
 }
 
+// TaskRunSucceededReason is an enum used to store all TaskRun reason for
+// the Succeeded condition that are controlled by the TaskRun itself. Failure
+// reasons that emerge from underlying resources are not included here
+type TaskRunSucceededReason string
+
+const (
+	// TaskRunSucceededReasonStarted is the reason set when the TaskRun has just started
+	TaskRunSucceededReasonStarted TaskRunSucceededReason = "Started"
+	// TaskRunSucceededReasonRunning is the reason set when the TaskRun is running
+	TaskRunSucceededReasonRunning TaskRunSucceededReason = "Running"
+	// TaskRunSucceededReasonSuccessful is the reason set when the TaskRun completed successfully
+	TaskRunSucceededReasonSuccessful TaskRunSucceededReason = "Succeeded"
+	// TaskRunSucceededReasonFailed is the reason set when the TaskRun completed with a failure
+	TaskRunSucceededReasonFailed TaskRunSucceededReason = "Failed"
+	// TaskRunSucceededReasonCancelled is the reason set when the Taskrun is cancelled by the user
+	TaskRunSucceededReasonCancelled TaskRunSucceededReason = "TaskRunCancelled"
+	// TaskRunSucceededReasonTimedOut is the reason set when the Taskrun has timed out
+	TaskRunSucceededReasonTimedOut TaskRunSucceededReason = "TaskRunTimeout"
+)
```

Comment:
Similar NIT as above, if we can rename `TaskRunSucceededReason` to `TaskRunReason`, `TaskRunSucceededReasonStarted` to `TaskRunReasonStarted` ...

Nicely collected all reasons under `types` 🙏